Duplicati/Library/Backend/OAuthHelper/JSONWebHelper.cs

```diff
@@ -150,7 +150,28 @@ public virtual async Task<HttpWebResponse> PostMultipartAsync(string url, Action
                 await rs.WriteAsync(bodyTerminator, 0, bodyTerminator.Length, cancelToken).ConfigureAwait(false);
             }
 
-            return (HttpWebResponse)(await req.GetResponseAsync().ConfigureAwait(false));
+            try
+            {
+                return (HttpWebResponse)(await req.GetResponseAsync().ConfigureAwait(false));
+            }
+            /*
+             * Catch any web exceptions and grab the error detail from the respose stream
+             */
+            catch (WebException wex)
+            {
+                using (Stream exstream = wex.Response.GetResponseStream())

```

Comment:
I don't think you can guarantee there to be a response stream present? If the error is "Network disconnected" there will be no response stream and the user will get an exception here.